tests/test_pyopener.py

```diff
@@ -272,3 +273,52 @@ def test_opener_fsspec_fs_tiff_threads_2():
             assert profile["driver"] == "GTiff"
             assert profile["count"] == 3
             assert src.read().shape == (3, 718, 791)
+
+
+def test_opener_multi_range_read():
+    """Test with Opener with multi-range-read method."""
+    from rasterio.abc import MultiByteRangeResourceContainer
+
+    class CustomResource(io.FileIO):
+        """Custom FileIO FS with `read_multi_range` method."""
+
+        def get_byte_ranges(
+            self,
+            offsets,
+            sizes,
+        ):
+            warnings.warn("Using MultiRange Reads", UserWarning, stacklevel=2)
+            return [
+                self._read_range(offset, size) for (offset, size) in zip(offsets, sizes)
+            ]
+
+        def _read_range(self, offset, size):
+            _ = self.seek(offset)
+            return self.read(size)
```

Comment:
@vincentsarago your test implementation is really nice, it's so cool to see this work.